core/src/main/java/smile/manifold/UMAP.java

```diff
@@ -582,7 +582,7 @@ private static void optimizeLayout(double[][] embedding, double[] curve, SparseM
                 }
             }
 
-            logger.info(String.format("The learning rate at %3d iterations: %.5f", iter, alpha));
+            logger.info(String.format("The learning rate at %3d iterations: %.5f", alpha, iter));
```

Comment:
This is wrong. Note that the first string interpolation is `%3d` for iterations.